### PR TITLE
Constrain cccl to prevent simultaneous installation.

### PR DESCRIPTION
**Pull request by bdice**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Once `cccl` is merged (https://github.com/conda-forge/staged-recipes/pull/23722) and this PR is merged, I will also submit a repodata patch for previous versions of `cuda-cccl-impl`. Reference: https://github.com/conda-forge/cuda-cccl-impl-feedstock/issues/2#issuecomment-1684371385

### CODE REVIEW

**Comment by bdice** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,9 +34,14 @@ source:
...
 
 build:
-  number: 0
+  number: 1
   skip: true  # [osx]
 
+requirements:
+  run_constrained:
+    # Prevent clobbering with cccl
+    - cccl <0a0
```
Comment (adibbley) [inline recipe/meta.yaml]: In [thrust PR](https://github.com/conda-forge/thrust-feedstock/pull/18/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9aR27) we used this syntax:
```cuda-cccl <0.0.0a0```
Should we follow that? I'm honestly not too familiar with this syntax.

Comment (bdice) [reply recipe/meta.yaml]: Shouldn't matter as far as I am aware. This is what @jakirkham suggested here: https://github.com/conda-forge/staged-recipes/pull/23722#discussion_r1298815269

**Comment by bdice** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (bdice) [reply recipe/meta.yaml]: I decided to match the previous syntax.
```suggestion
    - cccl <0.0.0a0
```

**Comment by bdice** [comment]
CI is passing now. @adibbley Can you review and merge if all looks good?

**Comment by jakirkham** [comment]
FWIW LGTM

**Comment by jakirkham** [comment]
Thanks Bradley & Alex! 🙏

**Pull request merged**